ndn/uri.go

```diff
@@ -28,22 +28,24 @@ const fdPattern = `^(?P<scheme>fd)://(?P<fd>[0-9]+)$`
 const ipv4Pattern = `^((25[0-4]|2[0-4][0-9]|1[0-9][0-9]|[0-9][0-9]|[0-9])\.){3}(25[0-4]|2[0-4][0-9]|1[0-9][0-9]|[0-9][0-9]|[0-9])$`
 const macPattern = `^(([0-9a-fA-F]){2}:){5}([0-9a-fA-F]){2}$`
 const udpPattern = `^(?P<scheme>udp[46]?)://\[?(?P<host>[0-9A-Za-z\:\.\-]+)(%(?P<zone>[A-Za-z0-9\-]+))?\]?:(?P<port>[0-9]+)$`
+const tcpPattern = `^(?P<scheme>tcp[46]?)://\[?(?P<host>[0-9A-Za-z\:\.\-]+)(%(?P<zone>[A-Za-z0-9\-]+))?\]?:(?P<port>[0-9]+)$`
 const unixPattern = `^(?P<scheme>unix)://(?P<path>[/\\A-Za-z0-9\:\.\-_]+)$`
 
 const (
-	unknownURI  uriType = iota
-	devURI      uriType = iota
-	ethernetURI uriType = iota
-	fdURI       uriType = iota
-	internalURI uriType = iota
-	nullURI     uriType = iota
-	udpURI      uriType = iota
-	unixURI     uriType = iota
+	unknownURI  URIType = iota
+	devURI      URIType = iota
+	ethernetURI URIType = iota
+	fdURI       URIType = iota
+	internalURI URIType = iota
+	nullURI     URIType = iota
+	udpURI      URIType = iota
+	tcpURI      URIType = iota
+	unixURI     URIType = iota
 )
 
 // URI represents a URI for a face.
 type URI struct {
-	uriType uriType
+	URIType URIType
```

Comment:
A member starting with a Capital letter is public, which may be modified from outside.